src/yoti_common/anchor.processor.js

```diff
@@ -0,0 +1,152 @@
+  this.associatedSource = anchorObj.associatedSource;
+};
+
+AttributeAnchor.prototype = {
+  getValue() { return this.value; },
+  getArtifactSignature() { return this.artifactSignature; },
+  getSubType() { return this.subType; },
+  getSignature() { return this.signature; },
+  getSignedTimeStamp() { return this.signedTimeStamp; },
+  getOriginServerCerts() { return this.originServerCerts; },
+  getAssociatedSource() { return this.associatedSource; },
+};
+
+module.exports.AnchorProcessor = class AnchorProcessor {
+  /**
+   * Extract matching Attribute Anchors from list.
+   *
+   * @param anchors
+   *
+   * @returns {Array}
+   */
+  static process(anchors) {
+    // This will contain a list of anchor values as objects
+    const anchorsData = {
+      sources: [],
+      verifiers: [],
+      processedValues: { // This is used to make sure the anchors values are unique
+        sources: [],
+        verifiers: [],
+      },
+    };
+
+    let originAnchorObj = {
+      value: '',
+      artifactSignature: '',
+      subType: '',
+      signedTimeStamp: '',
+      originServerCerts: '',
+      associatedSource: '',
+    };
+
+    for (let i = 0; i < anchors.length; i += 1) {
+      const anchor = anchors[i];
+      const certificatesList = anchor.originServerCerts;
+      originAnchorObj = Object.assign(originAnchorObj, anchor);
+
+      for (let n = 0; n < certificatesList.length; n += 1) {
+        const certArrayBuffer = certificatesList[n];
+        const certificateObj = AnchorProcessor.convertCertToX509(certArrayBuffer);
+        const extensionsData = certificateObj.extensions;
+        const anchorTypes = AnchorProcessor.getAnchorTypes();
+
+        Object.keys(anchorTypes).forEach((key) => {
+          const oidIndex = AnchorProcessor.findOidIndex(extensionsData, { id: anchorTypes[key] });
+          if (oidIndex !== -1) {
+            const anchorObj = extensionsData[oidIndex];
+            const anchorValue = anchorObj.value;
+            // Convert Anchor value from ASN.1 format to an object
+            const anchorValueAsn1 = forge.asn1.fromDer(anchorValue.toString('binary'));
+            if (anchorValueAsn1) {
+              const anchorParsedValue = anchorValueAsn1.value[0].value;
+              // Make sure the anchor values are unique
+              if (!anchorsData.processedValues[key].includes(anchorParsedValue)) {
+                originAnchorObj.value = anchorParsedValue;
+                const originServerCerts = originAnchorObj.originServerCerts;
+                const serverX509Certs = AnchorProcessor.convertCertsListToX509(originServerCerts);
+                originAnchorObj.originServerCerts = serverX509Certs;
+                anchorsData[key].push(new AttributeAnchor(originAnchorObj));
+                anchorsData.processedValues[key].push(originAnchorObj.value);
+              }
+            }
+          }
+        }, anchorsData); // End Object loop
+      } // End for loop
+    } // End for loop
```

Comment:
When you are looking at this in a small window `bracket highlight` wouldn't help.
But I am happy to remove them.

Reply:
I'm just thinking whether we set a precedent - we wouldn't want to do this for every loop, but then where do we draw a line - when there's 2 or 3 brackets in consecutive lines? In my opinion it would be better to not have them at all. Feel free to leave as is though if you think it helps a lot